staging/src/k8s.io/kubectl/pkg/cmd/events/event_flags.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+// PrintFlags composes common printer flag structs
+// used in the Event command.
+type PrintFlags struct {
```

Comment:
events command has custom printing mechanism unlike to other commands and this printing mechanism includes 3 flags(`--output-watch-events`, `--no-headers`, `--all-namespaces`). What is the preferred way to manage this default event printing by also supporting json and yaml printing?. Because only other command has such requirement is `get` command and apparently it is not achieving this in a preferred way.

Reply:
1. `--no-headers` from what I see is already supported in cli-runtime, see https://github.com/kubernetes/kubernetes/blob/7a8f8a32092ede759519f600ca4ac1a319a0bc1d/staging/src/k8s.io/cli-runtime/pkg/printers/interface.go#L42
2. `--all-namespaces` is a functionality of a command not a printer, it tells to go through all namespaces. Also I'm inclined to skip it for `kubectl events` since I don't see much value for it. We can always add it later if needed.
3. `--output-watch-events` that one is actually related with printing, but as I mentioned elsewhere, I'd skip it just like `--all-namespaces` for now. We can revisit that decision at a later time.

Reply:
`kubectl alpha events --all-namespaces` works today; I certainly use it.

`--output-watch-events` is more debatable, since `kubectl alpha events` deliberately ignores delete events; this was one of the original rationales for having a separate command.

Reply:
I removed `--output-watch-events` and custom print flags altogether. If user specifies `output` format, we'll use jsonyaml printing. If user does not specify anything, we'll fallback to current custom event printing.

Reply:
> `kubectl alpha events --all-namespaces` works today; I certainly use it.

I can be persuaded to include that particular one, like I said, that's definitely not one of the blocker on this.